src/shogun/clustering/KMeans.cpp

```diff
@@ -77,35 +77,31 @@ void CKMeans::set_initial_centers(SGMatrix<float64_t> centers)
 
 void CKMeans::set_random_centers(SGVector<float64_t> weights_set, SGVector<int32_t> ClList, int32_t XSize)
 {
+	/*initialise the cluster centroids randomly among all data*/
 	CDenseFeatures<float64_t>* lhs=
```

Comment:
we have a new feature in the code that does 
```auto lhs = ....```
You can use this here to make the code easier to read. It is mostly syntax sugar
@lisitsyn we can use this in Shogun right? Any checks needed....?